src/interactivity/base/HostSignalInputThread.cpp

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "HostSignalInputThread.hpp"
+#include "../inc/HostSignals.hpp"
+
+#include "../interactivity/inc/ServiceLocator.hpp"
+
+using namespace Microsoft::Console;
+using namespace Microsoft::Console::Interactivity;
+
+// Constructor Description:
+// - Creates the PTY Signal Input Thread.
+// Arguments:
+// - hPipe - a handle to the file representing the read end of the Host Signal pipe.
+HostSignalInputThread::HostSignalInputThread(_In_ wil::unique_hfile&& hPipe) :

```

Comment:
We don't need `_In_` in modern C++ anymore, do we?
Also I think I need to change my opinion: I previously agreed with Dustin that `&&` makes the intent of ownership transfer clear, but now that I read the code I feel like a version without any references at all is cleaner. It properly works even with prvalues, like when you want to write `new HostSignalInputThread(wil::make_unique<>(...))` (if that existed), etc.
I'm not sure if it can be argued in this particular case that the `&&` variant helps confer any additional meaning to future readers, what a handle called "unique" doesn't already.

Reply:
This is between you and @DHowett then. && already confuses me and you two understand it better than me. I just want whatever works.

Reply:
(I did drop SAL tho...)